exonum-java-binding-common/src/main/java/com/exonum/binding/common/message/BinaryTransactionMessage.java

```diff
@@ -127,4 +120,16 @@ public String toString() {
     return Arrays.toString(rawTransaction.array());
   }
 
+  /**
+   * Returns bytes by slicing raw transaction to avoid changing it's position.
+   */
+  private byte[] getBytes(int startOffset, int size) {
+    byte[] bytes = new byte[size];
+    ByteBuffer slice = rawTransaction.slice();
```

Comment:
Nit: `duplicate` will be a better fit because `slice` might generally have a smaller capacity than the original buffer, therefore, `position` won't work correctly. It won't happen in this case as position is suppossed to be always equal to 0, but one still have to think about it.